Split Makefile to be self-contained per directory - resubmit

linuxkidd opened the pull request:
Re-submit for accidentally closed PR #545 
## I think this addresses all the comments and concerns from that earlier PR.  Please let me know if you find anything else that needs attention.

Split the `src/Makefile` out to each directory where installation actions are needed:
- config_repo
- notification_images
- scripts
- src
- Top-level

The `Makefile` in each sub-directory can be called individually and it'll perform the proper actions, independent of other directories.

The top-level `Makefile` calls all sub-dir's functions as needed.
Tweaked `install.sh` to use the top-level `Makefile`.

EricClaeys reviewed (commented):
Michael, in Makefile, you added
```
	@if [ $(PKGBUILD) -eq 1 ]; then \
	  [ ! -e $(DESTDIR)$(libexecdir) ] && mkdir -p $(DESTDIR)$(libexecdir) \
	  install allsky.sh.repo $(DESTDIR)$(libexecdir)/allsky.sh; \
        fi
```
That code seems to be at odds with the title of the commit (Relocated allsky.sh back to top level, removed .repo version).  The code puts allsky.sh in libexecdir and uses the .repo version.
Are we keeping allsky.sh.repo or not?  Where will allsky.sh exist?

ckuethe commented on src/Makefile:
```diff
@@ -92,90 +107,50 @@ sunwait:
 	@$(MAKE) -C sunwait-src
 	@cp sunwait-src/sunwait .
-	@echo `date +%F\ %R:%S` Build of sunwait complete.
+	@echo `date +%F\ %R:%S` Done.
```
To be a little more like the previous behavior, could I get you to create some symlinks in the parent directory? Then when I am in the top level directory and type `make` I get some executables that I could run.

linuxkidd replied:
right now, the src/Makefile copies the binaries up to the ALLSKY directory.  I can change that behavior to be symlinks if you think that's better.

I guess that does ensure the top level is always the latest built binaries in use.

linuxkidd commented:
> Michael, in Makefile, you added
> ...
> That code seems to be at odds with the title of the commit (Relocated allsky.sh back to top level, removed .repo version). The code puts allsky.sh in libexecdir and uses the .repo version. Are we keeping allsky.sh.repo or not? Where will allsky.sh exist?

@EricClaeys  The `.repo` is a typo, I've fixed that in the next commit.  The 'libexec' stuff is only triggered when using the PKGBUILD environment variable -- not the default action.  It's for future package building convenience.

EricClaeys commented:
config.sh.repo:  Since config.sh is sourced after variables.sh, the following lines in config.sh.repo can use $ALLSKY_CONFIG instead of $ALLSKY_HOME/config:

```
CAMERA_SETTINGS_DIR="${ALLSKY_HOME}/config"
source "$ALLSKY_HOME/config/autocam.sh"
```

config_repo/Makefile: line 60: `@sed -e "s|User=pi|User=allsky|g" -e "s|/home/pi/allsky|$(bindir)|g" allsky.service.repo > allsky.service` Will changing the User to "allsky" break anything?
config_repo/Makefile: line 63: `@echo -e "export ALLSKY_TMP=/tmp\nexport ALLSKY_CONFIG=$(DESTDIR)$(sysconfdir)/allsky\nexport ALLSKY_SCRIPTS=$(DESTDIR)$(libexecdir)\nexport ALLSKY_NOTIFICATION_IMAGES=$(DESTDIR)$(sharedir)\nexport ALLSKY_IMAGES=/home/allsky/Pictures/" > $(DESTDIR)$(sysconfdir)/profile.d/allsky.sh`
Is that for possible future use?  The locations aren't what we're currently using.

notification_images/Makefile:  line 37:  `install -m 0644 *.jpg *.png $(DESTDIR)$(sharedir)/; \`  I created PR 565 to add AllskyNotifications.psd which I used to create all the .jpg and .png files, so it should probably be copied as well.

@linuxkidd, @ckuethe,
since this PR has lots of changes to almost every file, can we test it on a Pi prior to merging it, both as a clean install and as an upgrade (whatever that means)?  I suspect we missed a variable name change or location change or something...

Also, how do you put @ names in yellow?  When I do an @name it doesn't have a yellow background.

linuxkidd commented:
The yellow / brown or on the name is to highlight your own name so it's easier for you to spot.

Yes, I agree.. I'd much prefer you compile / test it.  It runs great for me with a fresh install on Buster.

EricClaeys commented:
I won't have access to my Pi for another week.

@ckuethe, are you able to install Michael's PR on your Pi and test it?  I think if it works for 2 people it should be good to go.

ckuethe commented:
Yeah, I'm reading through it now and will set up a test machine tomorrow.

linuxkidd commented:
Ok, I moved the `variables.sh` script back to $ALLSKY_HOME per Eric's request.

I also took the time to make an Uninstall function, including `uninstall.sh`

ckuethe commented on uninstall.sh:
```diff
@@ -0,0 +1,39 @@
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+if [[ $EUID -eq 0 ]]; then
```
I don't see the problem with running uninstall as root?
